apps/web/src/server/applications/case/s51/__tests__/applications-s51.test.js

```diff
@@ -415,6 +420,40 @@ describe('S51 Advice', () => {
 					expect(element.innerHTML).toContain('Check your answers');
 				});
 			});
+
+			describe('POST /case/123/project-documentation/21/s51-advice/create/check-your-answers', () => {
+				it('should display an error if data is not complete', async () => {
+					const response = await request.post(`${baseUrl}/create/check-your-answers`);
+					const element = parseHtml(response.text);
+
+					expect(element.innerHTML).toMatchSnapshot();
+					expect(element.innerHTML).toContain('An error occurred, please try again later');
+				});
+
+				it('should go to properties page showing a display banner if data is complete', async () => {
+					nock('http://test/').post('/applications/s51-advice').reply(200, { id: 1 });
+
+					const response = await request.post(`${baseUrl}/create/check-your-answers`).send({
+						title: 'S51 title',
+						enquirer: 's51 enquirer',
+						firstName: 's51 first name',
+						lastName: 's51 lastname',
+						enquiryMethod: 'email',
+						enquiryDate: '2020-01-01',
+						enquiryDetails: 's51 enquiry details',
+						adviser: 's51 adviser',
+						adviceDate: '2021-01-01',
+						adviceDetails: 's51 advice details'
+					});
+
+					expect(response?.headers?.location).toEqual('../../s51-advice/1/properties');
```

Comment:
I don't know if you want to check this explicitly, if not you can add `.redirects(1);` to the request to follow the redirect.

Reply:
the ID contained in the redirecting URL comes from the processing of the advice. I suppose that checking whether it's correct or not, is an extra-check